packages/refine-react/src/tailwind/components/criterion-group/criterion-group.tsx

```diff
@@ -9,12 +9,14 @@ export type CriterionGroupProps = {
 };
 
 export const CriterionGroup = ({ index, criteria }: CriterionGroupProps) => {
-  const { blueprint, groupedBlueprint } = useQueryBuilder();
+  const { blueprint, groupedBlueprint, errors: allErrors } = useQueryBuilder();
   const group = groupedBlueprint[index];
 
   const addCriterion = () =>
     blueprint.insertCriterion(group[group.length - 1].position);
 
+  const errors = allErrors[index];
```

Comment:
I think I might have this wrong here actually. The shape of the errors object is:

```ts
{
  0: [
    {
      id: 23434,
      message: "You messed up big time",
    },
    {
      id: 454534,
      message: "Good luck with your life",
    },
  ],
}
```

Is the key of that object the index of the offending criterion? So for example it could have keys of `0`, `2`, and `3`, each of which has an array of errors for that criterion?

Currently in the Vue implementation it looks like it renders on every criterion group:

<img width="885" alt="Screenshot 2022-03-29 at 23 40 15" src="https://user-images.githubusercontent.com/23709612/160718501-a61efc11-775b-405f-b632-597fb4c64c92.png">

I think this is because in the Vue implementation it's getting errors for the index but the index it's using is the index within a criterion group rather than the entire flat blueprint.

Am I right in thinking that I want to render these errors above the offending criterion? And the of the array of errors is the index of the criterion, including conjunctions?

So for example in this case if there was a key of `2` with errors it would appear:
0 - criterion
1 - conjunction
2 - criterion       <--- here
3 - criterion